### PR TITLE
ID casing fixes

### LINKED ISSUE #18346: "azurerm_monitor_diagnostic_setting": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.22.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_diagnostic_setting

### Terraform Configuration Files

```hcl
resource "azurerm_monitor_diagnostic_setting" "diag_setting" {
  name                       = "audit-log"
  target_resource_id         = var.resourceid
  log_analytics_workspace_id = var.audit_log_law_id

  log {
    category = "AuditEvent"
    enabled  = true

    retention_policy {
      days    = 0
      enabled = false
    }
  }
  metric {
    category = "AllMetrics"
    enabled  = false
    retention_policy {
      enabled = false
    }
  }
}
```

### Debug Output/Panic Output

```shell
Error: parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
```

### Expected Behaviour

Should work.

Basically same issue as https://github.com/hashicorp/terraform-provider-azurerm/issues/18235.
But it seems it was fixed in 3.22.0 for `azurerm_log_analytics_solution` only. So filing an other issue for `azurerm_monitor_diagnostic_setting` resource which has the same problem which needs a fix as well.

### Actual Behaviour

New configuration (were persumumely no state is present) it  works but for existing configuration / deployment with an existing state terraform apply or plan fails with the error: "parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups""

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### LINKED ISSUE #18360: azurerm_security_center_workspace": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.22.0

### Affected Resource(s)/Data Source(s)

azurerm_security_center_workspace

### Terraform Configuration Files

```hcl
resource "azurerm_security_center_workspace" "workspace" {
  scope        = join("", ["/subscriptions/", data.azurerm_subscriptions.msdn.subscriptions.0.subscription_id])
  workspace_id = azurerm_log_analytics_workspace.workspace.id
}
```

### Debug Output/Panic Output

```shell
Error: parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
```

### Expected Behaviour

Should work.

Appears the same issue as [#18235 ](https://github.com/hashicorp/terraform-provider-azurerm/issues/18235).

### Actual Behaviour

Brand new configuration which fails to deploy or destroy.  Other security center resources are present in state and need removing via the CLI to destroy.

Error: Reading Security Center Log Analytics Workspace ID: parsing "/subscriptions/SUB_ID/resourcegroups/RG_NAME/providers/microsoft.operationalinsights/workspaces/WORKSPACE_NAME": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"

Present in state file as per below:

```hcl
tf state show azurerm_security_center_workspace.workspace
# azurerm_security_center_workspace.workspace: (tainted)
resource "azurerm_security_center_workspace" "workspace" {
    id           = "/subscriptions/SUB_ID/providers/Microsoft.Security/workspaceSettings/default"
    scope        = "/subscriptions/SUB_ID"
    workspace_id = "/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.OperationalInsights/workspaces/WORKSPACE_NAME"
}
```

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by favoretti**
`azurerm_monitor_diagnostic_settings`: fix casing parsing for workspace_id
`azurerm_security_center_workspace`: fix casing parsing for workspace_id

Fixes #18346
Fixes #18360

### CODE REVIEW

**Comment by favoretti** [comment]
image

**Comment by favoretti** [comment]
image
Failures are old and not related to the change.

File: internal/services/msi/user_assigned_identity_resource.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func resourceArmUserAssignedIdentityRead(d *pluginsdk.ResourceData, meta interfa
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := commonids.ParseUserAssignedIdentityID(d.Id())
+	id, err := commonids.ParseUserAssignedIdentityIDInsensitively(d.Id())
```
Comment (katbyte) [inline internal/services/msi/user_assigned_identity_resource.go:121]: I think we'll need to do a state migration here to ensure the ID stored in state is correct? like we did in #18291

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the fix @favoretti - 2 of the 3 LGTM and i just think we need to do a state migration for the 3 i've left a comment on

**Comment by favoretti** [comment]
@katbyte API returns is consistently incorrectly and it's all fixes for Read, no matter how we store it in state, response from the API will be wrong. 

Not sure state migration would help any in this case?

File: internal/services/msi/user_assigned_identity_resource.go (hunk 1 above)
Comment (katbyte) [reply internal/services/msi/user_assigned_identity_resource.go:121]: d.id is the resource ID? so much like #18291 we'll need to apply the same type of fix including a state migration for existing users insuring that the ID in state is in the correct case

Comment (favoretti) [reply internal/services/msi/user_assigned_identity_resource.go:121]: Ahh, I'm with stupid. I'll fix tomorrow. Next release then.. or the former 2 without this one. Not at my pc atm unfortunately

Comment (katbyte) [reply internal/services/msi/user_assigned_identity_resource.go:121]: it happens! i'll remove this bit from this PR so we can get the other two into todays release and leave the user assigned identity for next

Comment (katbyte) [reply internal/services/msi/user_assigned_identity_resource.go:121]:
```suggestion
	id, err := commonids.ParseUserAssignedIdentityID(d.Id())
```

**Review by katbyte — APPROVED** [review]
LGTM 🌩️

**Pull request merged**